app/sign-in/SignInForm.tsx

```diff
@@ -0,0 +1,88 @@
+//
+// This source file is part of the Stanford Biodesign Digital Health ENGAGE-HF open-source project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+'use client'
+import { signInWithEmailAndPassword, signInWithPopup } from 'firebase/auth'
+import { useTranslations } from 'next-intl'
+import { z } from 'zod'
+import { Button } from '@stanfordbdhg/design-system/components/Button'
+import { Input } from '@stanfordbdhg/design-system/components/Input'
+import {
+  Separator,
+  SeparatorText,
+} from '@stanfordbdhg/design-system/components/Separator'
+import { Field } from '@stanfordbdhg/design-system/forms/Field'
+import { useForm } from '@stanfordbdhg/design-system/forms/useForm'
+import { auth, authProvider } from '../../modules/firebase/clientApp'
+
+const formSchema = z.object({
+  email: z.string().min(1, 'Email is required'),
+  password: z.string().min(1, 'Password is required'),
+})
+
+export const SignInForm = () => {
+  const t = useTranslations()
+  const form = useForm({
+    formSchema,
+    defaultValues: { email: '', password: '' },
+  })
+
+  const handleSubmit = form.handleSubmit(async ({ email, password }) => {
+    try {
+      await signInWithEmailAndPassword(auth, email, password)
+    } catch (error) {
+      if (
+        error instanceof Error &&
+        'code' in error &&
+        error.code === 'auth/invalid-credential'
+      ) {
+        form.setFormError(t('signIn_formError_invalidCredentials'))
+      } else {
+        form.setFormError(t('signIn_formError_unknown'))
+      }
+    }
+  })
+
+  return (
+    <div className="mx-auto grid w-[350px] gap-4">
```

Comment:
For ENGAGE-HF, we might want to include Stanford SSO at a later point. For testing & demonstration purposes it is fine to keep the other providers in here for now, especially username and password with would be a great escape hatch around SSO for testers for now.

Reply:
Just to clarify, because I am not sure if I got it. 

1. For now we can leave as is
2. In the future we want Stanford SSO only

Is that true?

Reply:
Exactly; that's correct. We might want to keep the functionality here as we could nicely move it into the SpeziWebDashboardTemplate in the future.